### ISSUE TITLE
Check for four spaces for nested lists

### ISSUE THREAD

**Issue by koppor** [DavidAnson/markdownlint#284]
When mixing lists and enumarations, the sub list should have an indent of 3 spaces:

```markdown
1. example1

   - sub 11
   - sub 12

2. example2

   - sub 21
   - sub 22
```

In case one indents with four spaces, there is no error reported:

```markdown
1. example1

    - sub 11
    - sub 12

2. example

    - sub 21
    - sub 22
```

There should be a warning, that three spaces are enough.

Howver, two spaces are too less since that does not work, "because the nesting of two spaces is insufficient to push it into the ordered list item"). This is catched by [MD007](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md007---unordered-list-indentation) and thus shouldn't be an issue.

(This is a follow-up to https://github.com/DavidAnson/markdownlint/issues/20#issuecomment-621427480)
